### PR TITLE
[SYCL] Speed up atomic_ref tests

### PR DESCRIPTION
**Pull request by vladimirlaz**
The full test coverage of sycl::atomic_ref  functionality is reached in SYCL-CTS tests. Limiting tests to speed up test execution (600+ seconds to 30- seconds):
 - single build per test is done having all test cases;
 - limiting type coverage to float to use native float atomics (to caver native fp atomics if available) and int (for and, or, xor operation which does not support floating point operation);
 - guard deprecated API (::sycl::ext::oneapi::atomic_ref) with RUN_DEPRECATED and removed from default execution;
 - guard full type coverage with FULL_ATOMIC32_COVERAGE and FULL_ATOMIC64_COVERAGE macros;
 - fixed atomic_memory_scope_capabilities checks.

### CODE REVIEW

**Review by sergey-semenov — DISMISSED** [review]

File: SYCL/AtomicRef/add.h [hunk 1]
```diff
@@ -281,100 +287,44 @@ template <access::address_space space, typename T, typename Difference = T,
...
+  if (std::find(scopes.begin(), scopes.end(), memory_scope::system) !=
       scopes.end()) {
-    std::cout << "Skipping test\n";
-    return;
+    add_test<space, T, Difference, order, memory_scope::system>(q, N);
   }
-  add_test<space, T, Difference, order, memory_scope::system>(q, N);
-#elif defined(WORK_GROUP)
-  if (std::find(scopes.begin(), scopes.end(), memory_scope::system) ==
+  if (std::find(scopes.begin(), scopes.end(), memory_scope::system) !=
```
Comment (t4c1) [inline SYCL/AtomicRef/add.h]:
```suggestion
  if (std::find(scopes.begin(), scopes.end(), memory_scope::work_group) !=
```
Just noticed this is actually wrong.

File: SYCL/AtomicRef/add.h [hunk 2]
```diff
@@ -281,100 +287,44 @@ template <access::address_space space, typename T, typename Difference = T,
...
       scopes.end()) {
-    std::cout << "Skipping test\n";
-    return;
+    add_test<space, T, Difference, order, memory_scope::work_group>(q, N);
   }
-  add_test<space, T, Difference, order, memory_scope::work_group>(q, N);
-#elif defined(SUB_GROUP)
-  if (std::find(scopes.begin(), scopes.end(), memory_scope::system) ==
+  if (std::find(scopes.begin(), scopes.end(), memory_scope::system) !=
```
Comment (t4c1) [inline SYCL/AtomicRef/add.h]:
```suggestion
  if (std::find(scopes.begin(), scopes.end(), memory_scope::sub_group) !=
```

File: SYCL/AtomicRef/compare_exchange_generic_local.cpp [hunk 3]
```diff
@@ -135,13 +135,7 @@
...
 // RUN: %ACC_RUN_PLACEHOLDER %t.out
 
-// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_70 -DACQ_REL -DSYSTEM
-// RUN: %HOST_RUN_PLACEHOLDER %t.out
-// RUN: %GPU_RUN_PLACEHOLDER %t.out
-// RUN: %CPU_RUN_PLACEHOLDER %t.out
-// RUN: %ACC_RUN_PLACEHOLDER %t.out
-
-// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_70 -DACQ_REL -DSYSTEM -DATOMIC64
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_70
```
Comment (t4c1) [inline SYCL/AtomicRef/compare_exchange_generic_local.cpp:1]: This change is inconsistent with what is done in other files.

File: SYCL/AtomicRef/sub.h [hunk 4]
```diff
@@ -282,100 +288,45 @@ template <access::address_space space, typename T, typename Difference = T,
...
+  if (std::find(scopes.begin(), scopes.end(), memory_scope::work_group) !=
       scopes.end()) {
-    std::cout << "Skipping test\n";
-    return;
+    sub_test<space, T, Difference, order, memory_scope::work_group>(q, N);
   }
-  sub_test<space, T, Difference, order, memory_scope::work_group>(q, N);
-#elif defined(SUB_GROUP)
-  if (std::find(scopes.begin(), scopes.end(), memory_scope::system) ==
+  if (std::find(scopes.begin(), scopes.end(), memory_scope::system) !=
```
Comment (t4c1) [inline SYCL/AtomicRef/sub.h]:
```suggestion
  if (std::find(scopes.begin(), scopes.end(), memory_scope::sub_group) !=

```

File: SYCL/AtomicRef/xor_local.cpp [hunk 5]
```diff
@@ -3,145 +3,7 @@
 // separately. This is controlled by macros, defined by RUN commands. Defaults
 // (no macro for a group) are: 32 bit, relaxed and device.
```
Comment (t4c1) [inline SYCL/AtomicRef/xor_local.cpp]: You are changing this behavior, so this comment can be removed (from all cpp files).

**Review by t4c1 — CHANGES_REQUESTED** [review]
Is reducing the test coverage to a single type per operation necessary to reach the required testing time? If it is, it might make sense to put tests for other types behind a macro like you did with deprecated tests from `ext` namespace.

**Review by t4c1 — APPROVED** [review]
LGTM

**Comment by vladimirlaz** [comment]
@sergey-semenov could you please have a look/approve?

**Comment by vladimirlaz** [comment]
@yanfeng3721 could you please have a look?

**Comment by yanfeng3721** [comment]
> @yanfeng3721 could you please have a look?

LGTM.

**Pull request merged**